coordinator/core/core.go

```diff
@@ -168,41 +175,67 @@ func (c *Core) GetTLSCertificate(clientHello *tls.ClientHelloInfo) (*tls.Certifi
 	return util.TLSCertFromDER(c.cert.Raw, c.privk), nil
 }
 
-func (c *Core) loadState() (*x509.Certificate, *ecdsa.PrivateKey, error) {
+func (c *Core) loadState() (*x509.Certificate, []*x509.Certificate, *ecdsa.PrivateKey, error) {
 	stateRaw, err := c.sealer.Unseal()
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 	if len(stateRaw) == 0 {
-		return nil, nil, nil
+		return nil, nil, nil, nil
 	}
 
 	// load state
 	c.zaplogger.Info("applying sealed state")
 	var loadedState sealedState
 	if err := json.Unmarshal(stateRaw, &loadedState); err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 
 	// set Core to loaded state
 	cert, err := x509.ParseCertificate(loadedState.RawCert)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 	privk, err := x509.ParseECPrivateKey(loadedState.Privk)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 
 	if err := json.Unmarshal(loadedState.RawManifest, &c.manifest); err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 	c.rawManifest = loadedState.RawManifest
 
+	// Load update manifest if one has been set
+	if loadedState.RawUpdateManifest != nil {
+		if err := json.Unmarshal(loadedState.RawUpdateManifest, &c.updateManifest); err != nil {
+			return nil, nil, nil, err
+		}
+		c.rawUpdateManifest = loadedState.RawUpdateManifest
+	}
+
 	c.state = loadedState.State
 	c.activations = loadedState.Activations
 	c.secrets = loadedState.Secrets
-	return cert, privk, err
+
+	// Parse & write X.509 admin certificates from sealed state
+	adminCerts := make([]*x509.Certificate, len(c.updateManifest.Admins))
+	for _, value := range c.updateManifest.Admins {
+		block, _ := pem.Decode([]byte(value))
+		if err != nil {
+			c.zaplogger.Error("Could not decode specified admin client certificate from sealed state", zap.Error(err))
+			return nil, nil, nil, err
+		}
+		cert, err := x509.ParseCertificate(block.Bytes)
+		if err != nil {
+			c.zaplogger.Error("Could not parse specified admin client certificate from sealed state", zap.Error(err))
+			return nil, nil, nil, err
+		}
+
+		adminCerts = append(adminCerts, cert)
+	}
+
+	return cert, adminCerts, privk, err
```

Comment:
I think we don't need to return adminCerts. Just create it before the other assignments above and then assign it with the others.